### ISSUE TITLE
Unhandled Exception: System.InvalidCastException: Unable to cast object of type 'System.DateTime' to type 'System.Linq.Expressions.LambdaExpression'.

### ISSUE THREAD

**Issue by Chrille79** [dotnet/efcore#11021]
I got an app that uses queryfilter
builder.Entity<Blog>().HasQueryFilter(x => x.From < DateTime.Today);

This was working before i upgraded the nuget packages. But with the new version I have to use DateTime.Now

This is working if you use 
```<PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0" />```
but does not work if you use
```<PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.5" />```

```
Exception message:
Stack trace:
Unhandled Exception: System.InvalidCastException: Unable to cast object of type 'System.DateTime' to type 'System.Linq.Expressions.LambdaExpression'.
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateSetFilterParametersExpressions(ParameterExpression& contextVariableExpression)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateExecutorLambda[TResults]()
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](Expression query, INodeTypeProvider nodeTypeProvider, IDatabase database, IDiagnosticsLogger`1 logger, Type contextType)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass15_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at System.Linq.Queryable.Any[TSource](IQueryable`1 source)
   at ConsoleApp2.Program.Run() in C:\temp\ConsoleApp2\ConsoleApp2\Program.cs:line 21
   at ConsoleApp2.Program.Main(String[] args) in C:\temp\ConsoleApp2\ConsoleApp2\Program.cs:line 14
```

### Steps to reproduce
Create an console version and use 
 ```
<PropertyGroup>
    <PublishWithAspNetCoreTargetManifest>false</PublishWithAspNetCoreTargetManifest>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0" />
  </ItemGroup>
```

1) Publish! To see that the program works
2) Change to version 2.0.5
3) Publish! To see the error

```c#
using Microsoft.EntityFrameworkCore;
using System;
using System.Linq;

namespace ConsoleApp2
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");

            new Program().Run();
        }

        public void Run()
        {
            var db = new BloggingContext();
            db.Database.EnsureCreated();
            db.Blogs.Any();
            Console.WriteLine("Working");
            Console.ReadLine();
        }

        public class BloggingContext : DbContext
        {
            public DbSet<Blog> Blogs { get; set; }

            protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
            {
                optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=MyDatabase;Trusted_Connection=True;");
            }

            protected override void OnModelCreating(ModelBuilder builder)
            {
                //Works with DateTime.Now but not with DateTime.Today
                builder.Entity<Blog>().HasQueryFilter(x => x.From < DateTime.Today);
            }
        }

        public class Blog
        {
            public int Id { get; set; }
            public string Name { get; set; }
            public DateTime From { get; set; }
        }
    }
}

```

### Further technical details
EF Core version: Working in 2.0.0. Error in 2.0.5
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Win10
IDE: (e.g. Visual Studio 2017 15.4)

**Comment by ralmsdeveloper** [comment]
@Chrille79.
This has been fixed in version 2.1!

But you can solve this by switching **DateTime.Today** to **DateTime.Now**

`builder.Entity<Blog>().HasQueryFilter(x => x.From < DateTime.Now);`

**Comment by smitpatel** [comment]
duplicate of #10271
